app/flame_chart/profile_model.ts

```diff
@@ -85,8 +90,21 @@ export function buildThreadTimelines(events: TraceEvent[], { visibilityThreshold
       event.dur &&
       event.dur > visibilityThreshold
   );
-  events.sort(eventComparator);
+  
+  for (const [tid, threadName] of threadNameByTid.entries()) {
+    if (threadNameToMinTidMap.has(threadName)) {
+      threadNameToMinTidMap.set(threadName, Math.min(threadNameToMinTidMap.get(threadName), tid));
+    } else {
+      threadNameToMinTidMap.set(threadName, tid)
+    }
+  }
+  // Merge all of the threads with a single thread name under the same tid.
```

Comment:
nit: move this comment just above the for loop starting on line 94 (since that for loop is also doing some of the work to merge threads).